### PR TITLE
Do not error for job worker user task

### PR DESCRIPTION
**Pull request by barmac**
Two parts:
* https://github.com/camunda/bpmnlint-plugin-camunda-compat/commit/dd3173ee5f6fd0b5e306a174ce8fe3ee62b61030 removes correct rule due to requirements change
* https://github.com/camunda/bpmnlint-plugin-camunda-compat/commit/ce3f2001ae07b87501b851462f7a0865b231028b fixes the ordering bug which would mark correct user tasks as invalid

Related to https://github.com/camunda/camunda-modeler/issues/4690
Related to https://github.com/camunda/camunda-modeler/issues/4718

### CODE REVIEW

**Comment by barmac** [comment]
CC @misiekhardcore as you implemented the task listener rule (potential learning)

File: index.js [hunk 1]
```diff
@@ -249,7 +248,7 @@ module.exports = {
             }
 
             return type;
-          }, null)
+          }, null) || 'error'
```
Comment (philippfromme) [inline index.js:251]: Why is this change necessary?

File: index.js [hunk 2]
```diff
@@ -96,7 +96,6 @@ const camundaCloud86Rules = withConfig({
 
 const camundaCloud87Rules = withConfig({
   ...omit(camundaCloud86Rules, [ 'no-task-listeners' ]),
-  'zeebe-user-task': 'error',
```
Comment (philippfromme) [inline index.js:99]: I would prefer to remove the rule and all related tests for now. We don't keep rules in the codebase that aren't used.

File: rules/camunda-cloud/task-listener.js [hunk 3]
```diff
@@ -17,18 +17,18 @@ module.exports = skipInNonExecutableProcess(function() {
...
-      reportErrors(node, reporter, errors);
-    }
-
     const taskListeners = findExtensionElement(node, 'zeebe:TaskListeners');
 
     if (!taskListeners) {
       return;
     }
 
+    let errors = hasExtensionElement(node, 'zeebe:UserTask');
```
Comment (philippfromme) [inline rules/camunda-cloud/task-listener.js:26]: This is still going to report the same error which only states that the extension element is required. It doesn't hint at the fact that this extension element is only required if another extension element is present. We have a similar error type for properties (cf. https://github.com/camunda/bpmnlint-plugin-camunda-compat/blob/main/rules/utils/error-types.js#L17). I'd consider adding a `EXTENSION_ELEMENT_DEPENDENT_REQUIRED` error type.

**Review by philippfromme — COMMENTED** [review]
A couple of comments.

File: index.js (hunk 1 above)
Comment (barmac) [reply index.js:251]: Otherwise, the `all` configuration contains `null` value for rules not present in the presets.

File: index.js (hunk 2 above)
Comment (barmac) [reply index.js:99]: I think a complete rule removal is a breaking change. I don't think it's necessary for the bugfix.

File: rules/camunda-cloud/task-listener.js (hunk 3 above)
Comment (barmac) [reply rules/camunda-cloud/task-listener.js:26]: I will use the error you suggested.

Comment (barmac) [reply rules/camunda-cloud/task-listener.js:26]: We have just clarified that we will remove the check since you cannot model a job worker user task with task listeners in the UI.

File: index.js (hunk 2 above)
Comment (philippfromme) [reply index.js:99]: I consider the rule being there a bug. We remove it, we fix the bug. 😆

File: rules/camunda-cloud/task-listener.js (hunk 3 above)
Comment (barmac) [reply rules/camunda-cloud/task-listener.js]: Done via https://github.com/camunda/bpmnlint-plugin-camunda-compat/pull/182/commits/df3d36366e052e0615318ef48676e436092d3c82

**Comment by barmac** [comment]
I've pushed `zeebe-user-task` removal via https://github.com/camunda/bpmnlint-plugin-camunda-compat/pull/182/commits/8a4716108ef67ecd57a8300ad9d37704c0421e81 and worded it in a non-breaking-change way. WDYT?
@nikku could you have a look too?

**Comment by nikku** [comment]
Works for me. This is supporting future platforms, rules are "a floating target" for these. As long as we don't break API we can phrase things as _minor_.

**Comment by philippfromme** [comment]
Good job! 👍🏻
